### PR TITLE
RadioButtons

### PR DESCRIPTION
**Pull request by skandarhayek**

### CODE REVIEW

File: src/framework/radioButtons/RadioButtons.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  className: PropTypes.string,
+  elements: PropTypes.arrayOf(PropTypes.shape({
+    label: PropTypes.string,
+    value: PropTypes.any,
+  })).isRequired,
+  onSelect: PropTypes.func.isRequired,
+  selectedElement: PropTypes.shape({
+    label: PropTypes.string,
+    value: PropTypes.any,
+  }),
```
Comment (mariacasancho) [inline src/framework/radioButtons/RadioButtons.jsx]: you need to add name here. I get an error: 
47:30  error  'name' is missing in props validation

File: src/framework/organizationSwitcher/_organizationSwitcher.scss [hunk 2]
```diff
@@ -9,7 +9,7 @@ body {
 }
 
 .organizationSwitcher {
-  border: 1px solid #d4dadc;
+  border: 1px solid $borderGrey;
```
Comment (mariacasancho) [inline src/framework/organizationSwitcher/_organizationSwitcher.scss:12]: order

File: src/framework/radioButtons/RadioButtons.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        const testCase = TestCaseFactory.create(RadioButtons, props);
+        expect(
+          testCase.find('.radioButtons--element').length
+        ).toEqual(2);
+      });
+    });
+  });
+
+  describe('Actions', () => {
+    describe('upon clicking of a option', () => {
```
Comment (mariacasancho) [inline src/framework/radioButtons/RadioButtons.spec.jsx]: should we have a test to check that contains the active class?

**Review by mariacasancho — CHANGES_REQUESTED** [review]
just small comments, good job!

File: src/framework/radioButtons/RadioButtons.jsx (hunk 1 above)
Comment (arijitdasgupta) [reply src/framework/radioButtons/RadioButtons.jsx]: Thanks! Good spot. Done.

File: src/framework/radioButtons/RadioButtons.jsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    this.setState({
+      selectedOption: element,
+    });
+    this.props.onSelect(element.value);
+  }
+
+  render() {
+    const onSelectCreator = value => () => this.onSelectOption(value);
+
+    const radioElements = this.props.elements.map((element, index) => {
```
Comment (DarioG) [inline src/framework/radioButtons/RadioButtons.jsx]: What about creating another RadioElement component?

File: src/framework/radioButtons/RadioButtons.jsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+  onSelectOption(element) {
+    this.setState({
+      selectedOption: element,
+    });
+    this.props.onSelect(element.value);
+  }
+
+  render() {
+    const onSelectCreator = value => () => this.onSelectOption(value);
```
Comment (DarioG) [inline src/framework/radioButtons/RadioButtons.jsx]: If we create the RadioElement component we wouldnt have to create this hacky wrapper

File: src/framework/radioButtons/RadioButtons.jsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+      // https://reactjs.org/docs/react-component.html#componentdidmount
+      // eslint-disable-next-line react/no-did-mount-set-state
+      this.setState({
+        selectedOption: this.props.selectedElement
+      });
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.state.selectedOption !== this.props.selectedElement) {
```
Comment (DarioG) [inline src/framework/radioButtons/RadioButtons.jsx]: You have 2 methods duplicated here.... this is a common symptom of not having a single source of truth....in this case for the selectedElement....If you ask me i would remove completely the state of the component and leave it up to the client to keep track of the selected element......

**Review by DarioG — COMMENTED** [review]
I added some comments....besides I think the test suite is missing a few cases that, in my opinion, should be tested

File: src/framework/radioButtons/RadioButtons.jsx (hunk 6 above)
Comment (arijitdasgupta) [reply src/framework/radioButtons/RadioButtons.jsx]: This was done this way to keep consistency with other components in the UI Framework, such as `Dropdown`. But totally I agree with you in general. 👍

**Review by mariacasancho — APPROVED** [review]

**Pull request merged**